sktime/transformations/compose.py

```diff
@@ -51,32 +51,80 @@ class TransformerPipeline(BaseTransformer, _HeterogenousMetaEstimator):
 
     Parameters
     ----------
-    transformers : list of sktime transformers, or
+    steps : list of sktime transformers, or
         list of tuples (str, transformer) of sktime transformers
         these are "blueprint" transformers, states do not change when `fit` is called
 
     Attributes
     ----------
-    transformers_ : list of tuples (str, transformer) of sktime transformers
-        clones of transformers in `transformers` which are fitted in the pipeline
-        is always in (str, transformer) format, even if transformers is just a list
-        strings not passed in transformers are unique generated strings
-        i-th transformer in `transformers_` is clone of i-th in `transformers`
+    steps_ : list of tuples (str, transformer) of sktime transformers
+        clones of transformers in `steps` which are fitted in the pipeline
+        is always in (str, transformer) format, even if `steps` is just a list
+        strings not passed in `steps` are replaced by unique generated strings
+        i-th transformer in `steps_` is clone of i-th in `steps`
+
+    Examples
+    --------
+    >>> # we'll construct a pipeline from 2 transformers below, in three different ways
+    >>> # preparing the transformers
+    >>> from sktime.transformations.series.exponent import ExponentTransformer
+    >>> t1 = ExponentTransformer(power=2)
+    >>> t2 = ExponentTransformer(power=0.5)
+
+    >>> # Example 1: construct without strings
+    >>> tp = TransformerPipeline(steps = [t1, t2])
+    >>> # unique names are generated for the two components t1 and t2
+    >>> tp.get_params()
+    {'steps': [ExponentTransformer(power=2), ExponentTransformer()],
+    'ExponentTransformer_1': ExponentTransformer(power=2),
+    'ExponentTransformer_2': ExponentTransformer(),
+    'ExponentTransformer_1__offset': 'auto',
+    'ExponentTransformer_1__power': 2,
+    'ExponentTransformer_2__offset': 'auto',
+    'ExponentTransformer_2__power': 0.5}
```

Comment:
are the names automatically created in upper camel case? If so, I would suggest to lower, so we would get `exponenttransformer_1`

Reply:
hm, usually names are in lower case 🤔